custom_model_runner/datarobot_drum/resource/predict_mixin.py

```diff
@@ -29,8 +30,19 @@ class PredictMixin:
 
     """
 
-    def do_predict(self, logger=None):
+    def _predict_or_transform(self, logger=None, predict=True):
         response_status = HTTP_200_OK
+        if predict is False and self._target_type != TargetType.TRANSFORM.value:
```

Comment:
As I understand, we should also check here for the case:
predict==True and target_type == TRANSFORM

Instead of this we could check for proper target_type in `do_transform` and `do_predict` methods and then use `_predict_or_transform` without additional param `predict`. And do resolution on target_type as we already checked it.

Or you just don't need `do_transform` and `do_predict` and can use `_predict_or_transform` directly from the server code.

Reply:
good call!